vertical-pod-autoscaler/pkg/recommender/model/vpa_test.go

```diff
@@ -142,3 +142,166 @@ func TestUpdateConditions(t *testing.T) {
 		})
 	}
 }
+
+func TestUpdateRecommendation(t *testing.T) {
+	cases := []struct {
+		name           string
+		containers     []string
+		recommendation *vpa_types.RecommendedPodResources
+		expectedLast   map[string]corev1.ResourceList
+	}{
+		{
```

Comment:
A test I'm missing here is a recommendation update - calling UpdateRecommendation with data similar to 2nd TC on the output of 1st TC; we should check that the test-container has updated recommendation and second-container keeps the old one.

Reply:
Good point, added.